### PR TITLE
Resuming segment downloading.

### PR DESCRIPTION
**Pull request by i-zolotarenko**

### CODE REVIEW

**Review by mrlika — CHANGES_REQUESTED** [review]

File: packages/p2p-media-loader-core/src/core.ts [hunk 1]
```diff
@@ -28,6 +28,8 @@ export class Core<TStream extends Stream = Stream> {
     p2pNotReceivingBytesTimeoutMs: 1000,
     p2pLoaderDestroyTimeoutMs: 30 * 1000,
     httpNotReceivingBytesTimeoutMs: 1000,
+    maxHttpFailedDownloadAttempts: 3,
+    maxPeerNotReceivingBytesTimeoutErrors: 3,
```
[the remark this answers is not in the document]
Comment (i-zolotarenko) [reply packages/p2p-media-loader-core/src/core.ts]: Peer bytes mismatch error will be thrown.

File: packages/p2p-media-loader-core/src/http-loader.ts [hunk 2]
```diff
@@ -1,70 +1,139 @@
...
+      } else {
+        throw new RequestError("http-error", response.statusText);
+      }
     }
-    requestControls.completeOnSuccess();
-  } catch (error) {
+
+    const { byteRange } = this;
+    if (byteRange) {
+      if (response.status === 200) {
```
Comment (mrlika) [inline packages/p2p-media-loader-core/src/http-loader.ts:85]:
```suggestion
      if (response.status === 200 && !request.segment.byteRange) {
```

Comment (mrlika) [inline packages/p2p-media-loader-core/src/http-loader.ts:85]:
```suggestion
      if (response.status === 200 && request.segment.byteRange) {
```

File: packages/p2p-media-loader-core/src/hybrid-loader.ts [hunk 3]
```diff
@@ -187,7 +189,13 @@ export class HybridLoader {
           this.requests.remove(request);
           break;
       }
-      request.markCheckedByProcessQueue();
+      request.markHandledByProcessQueue();
+      if (
+        now - request.failedAttempts.lastClearTimestamp >
```
Comment (mrlika) [inline packages/p2p-media-loader-core/src/hybrid-loader.ts:194]: Use last failed attempt timestamp

**Review by mrlika — APPROVED** [review]

**Pull request merged**